workflow/controller/controller.go

```diff
@@ -510,19 +510,61 @@ func (wfc *WorkflowController) processNextPodCleanupItem(ctx context.Context) bo
 				return err
 			}
 		case labelPodCompleted:
+			patch := []common.Operation{
+				{
+					Operation: "replace",
+					Path:      "/metadata/labels/workflows.argoproj.io~1completed",
+					Value:     "true",
+				},
+			}
+			obj, _, _ := wfc.podInformer.GetIndexer().GetByKey(namespace + "/" + podName)
+			if p, ok := obj.(*apiv1.Pod); ok {
```

Comment:
It feels like this is repeated 3 times here, and could be commoned up into a function.